polygeist/test/polygeist-opt/sycl/detect-reduction-accessor-versioning.mlir

```diff
@@ -25,30 +25,22 @@
 // CHECK-SAME:    %arg0: memref<?x!sycl_accessor_1_i32_rw_gb, 4>, %arg1: memref<?x!sycl_accessor_1_i32_r_gb, 4>) {
 
 // COM: Obtain a pointer to the beginning of the accessor %arg0.
-// CHECK:         %alloca_8 = memref.alloca() : memref<1x!sycl_id_1_>
-// CHECK-NEXT:    %c0_9 = arith.constant 0 : index
-// CHECK-NEXT:    %c0_i32 = arith.constant 0 : i32
-// CHECK-NEXT:    %2 = sycl.id.get %alloca_8[%c0_i32] : (memref<1x!sycl_id_1_>, i32) -> memref<?xindex>
-// CHECK-NEXT:    memref.store %c0_7, %2[%c0_9] : memref<?xindex>
-// CHECK-NEXT:    [[ARG0_BEGIN:%.*]] = sycl.accessor.subscript %arg0[%alloca_8] : (memref<?x!sycl_accessor_1_i32_rw_gb, 4>, memref<1x!sycl_id_1_>) -> memref<?xi32, 1>
+// CHECK:         %2 = sycl.id.constructor(%c0_7) : (index) -> memref<1x!sycl_id_1_>
+// CHECK-NEXT:    [[ARG0_BEGIN:%.*]] = sycl.accessor.subscript %arg0[%2] : (memref<?x!sycl_accessor_1_i32_rw_gb, 4>, memref<1x!sycl_id_1_>) -> memref<?xi32, 1>
 
 // COM: Obtain a pointer to the end of the accessor %arg0.
 // CHECK-NEXT:    %4 = sycl.accessor.get_range(%arg0) : (memref<?x!sycl_accessor_1_i32_rw_gb, 4>) -> !sycl_range_1_
-// CHECK-NEXT:    %alloca_10 = memref.alloca() : memref<1x!sycl_range_1_>
-// CHECK-NEXT:    %c0_11 = arith.constant 0 : index
-// CHECK-NEXT:    memref.store %4, %alloca_10[%c0_11] : memref<1x!sycl_range_1_>
-// CHECK-NEXT:    %c1_12 = arith.constant 1 : index
-// CHECK-NEXT:    %c0_i32_13 = arith.constant 0 : i32
-// CHECK-NEXT:    %5 = sycl.range.get %alloca_10[%c0_i32_13] : (memref<1x!sycl_range_1_>, i32) -> index
-// CHECK-NEXT:    %alloca_14 = memref.alloca() : memref<1x!sycl_id_1_>
-// CHECK-NEXT:    %c0_15 = arith.constant 0 : index
-// CHECK-NEXT:    %c0_i32_16 = arith.constant 0 : i32
-// CHECK-NEXT:    %6 = sycl.id.get %alloca_14[%c0_i32_16] : (memref<1x!sycl_id_1_>, i32) -> memref<?xindex>
-// CHECK-NEXT:    memref.store %5, %6[%c0_15] : memref<?xindex>
-// CHECK-NEXT:    [[ARG0_END:%.*]] = sycl.accessor.subscript %arg0[%alloca_14] : (memref<?x!sycl_accessor_1_i32_rw_gb, 4>, memref<1x!sycl_id_1_>) -> memref<?xi32, 1>
+// CHECK-NEXT:    %alloca_8 = memref.alloca() : memref<1x!sycl_range_1_>
+// CHECK-NEXT:    %c0_9 = arith.constant 0 : index
+// CHECK-NEXT:    memref.store %4, %alloca_8[%c0_9] : memref<1x!sycl_range_1_>
+// CHECK-NEXT:    %c1_10 = arith.constant 1 : index
+// CHECK-NEXT:    %c0_i32 = arith.constant 0 : i32
+// CHECK-NEXT:    %5 = sycl.range.get %alloca_8[%c0_i32] : (memref<1x!sycl_range_1_>, i32) -> index
+// CHECK-NEXT:    %6 = sycl.id.constructor(%5) : (index) -> memref<1x!sycl_id_1_>
+// CHECK-NEXT:    [[ARG0_END:%.*]] = sycl.accessor.subscript %arg0[%6] : (memref<?x!sycl_accessor_1_i32_rw_gb, 4>, memref<1x!sycl_id_1_>) -> memref<?xi32, 1>
 
-// CHECK:         [[ARG1_BEGIN:%.*]] = sycl.accessor.subscript %arg1[%alloca_18]
-// CHECK:         [[ARG1_END:%.*]] = sycl.accessor.subscript %arg1[%alloca_25]
+// CHECK:         [[ARG1_BEGIN:%.*]] = sycl.accessor.subscript %arg1[%8]
```

Comment:
Can we include the definition of these values in the test? Maybe use matching for them?

Reply:
How about changing them to `{{.*}}`? They were intentionally not included to simplify the checks, as they are the same codegen the get the begin and end of an accessor.